### ISSUE TITLE
I have a question about updating the deltas.

### ISSUE THREAD

**Issue by andreibelcin** [meownoid/tensorflow-rbm#10]
Hi, 

I have question about the f function in _initialize_vars(). I know that delta_w is equal to the learning rate times the difference between the positive and negative gradients (for momentum = 0) but I don't understand why you divide at the end with the size of x_new. I tried to remove it and I've noticed the error value exploding so I understand it's needed but I've never seen it in the mathematical model and I couldn't find any explanation online. Could you please explain why that term is needed?

Thanks a lot!

https://github.com/meownoid/tensorfow-rbm/blob/b55460e3fdbf83c2a9278c35cf20c6fed1d4666c/tfrbm/gbrbm.py#L27

**Comment by isaacgg** [comment]
Hi, I'm not the author, but I think he is doing this because when you compute deltas you actually have to compute the mean of the positive state - mean of the negative state, that would be x0*h0/batchsize - x1*h1/batchsize, otherwise big batches would produce big learning steps that would lead you to exploding gradients

**Comment by meownoid** [comment]
Thanks @isaacgg, that is correct!
